query-planner-js/src/buildPlan.ts

```diff
@@ -2473,86 +2582,55 @@ class FetchDependencyGraph {
         deferredGroups: allDeferredGroups,
       } = this.processRootMainGroups({
```

Comment:
Note that because `processRootMainGroups()` doesn't take the old `state`/return a new `state`, presumably when it eventually calls `updateForProcessedGroups()` for the `state` it internally creates, it won't alter the separate `state` in the upstream `processGroups()` call. This was the case before this PR too though, so it's probably fine to punt that to another PR if you'd like. (It might also be the case this was intentional.)